src/coord/src/persistcfg.rs

```diff
@@ -304,6 +304,24 @@ impl PersisterWithConfig {
                     envelope_details: crate::catalog::SerializedEnvelopePersistDetails::Upsert,
                 })
             }
+            SourceConnector::External {
+                connector: ExternalSourceConnector::Kafka(_),
+                envelope: SourceEnvelope::None(_),
+                ..
+            } if self.config.kafka_upsert_source_enabled => {
+                // NB: This gets written down in the catalog, so it should be
+                // safe to change the naming, if necessary. See
+                // Catalog::deserialize_item.
+                let name_prefix = format!("user-source-{}-{}", id, pretty);
```

Comment:
we can move the construction of primary_stream and timestamp_bindings_stream up before the match, right? IIUC all branches we'll ever add to this will want those two

Reply:
:+1: